compiler warnings

this may not matter, but for the record:

```
cc -o build/rtsprelay rtsprelay.c -g -O0 --std=c99 -Wall -pthread -I/usr/include/gstreamer-1.0 -I/usr/lib/x86_64-linux-gnu/gstreamer-1.0/include -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -lgstapp-1.0 -lgstrtsp-1.0 -lgstsdp-1.0 -lgio-2.0 -lgstrtspserver-1.0 -lgstbase-1.0 -lgstreamer-1.0 -lgobject-2.0 -lglib-2.0 
rtsprelay.c: In function 'on_options':
rtsprelay.c:223:14: warning: implicit declaration of function 'strdup' [-Wimplicit-function-declaration]
   ri->path = strdup(pathv[1]);
              ^
rtsprelay.c:223:12: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   ri->path = strdup(pathv[1]);
            ^
```
this is from building inside our Linux development environment

--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,8 @@
 BLDDIR = build
 TARGET = rtsprelay
 
-CFLAGS = -g -O0 --std=c99 -Wall
+CFLAGS = -g -O1 --std=c99 -D_XOPEN_SOURCE=600
+CFLAGS += -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers
 LIBS =
 
 PKGS = gstreamer-1.0 gstreamer-app-1.0 gstreamer-rtsp-1.0 gstreamer-rtsp-server-1.0

